wsf_scraping/pipelines.py

```diff
@@ -122,14 +122,20 @@ def check_keywords(self, item, spider_name, base_pdf_path):
     def process_item(self, item, spider):
         """Process items sent by the spider."""
 
-        base_pdf_path = os.path.join('/tmp', item['pdf'])
-        file_hash = get_file_hash(base_pdf_path)
+        if not item['pdf']:
+            raise DropItem(
+                'Empty filename, could not parse the pdf.'
+            )
+        file_hash = get_file_hash(item['pdf'])
         if self.database.is_scraped(file_hash):
             # File is already scraped in the database
             raise DropItem(
                 'Item footprint is already in the database'
             )
-        full_item = self.check_keywords(item, spider.name, base_pdf_path)
+        full_item = self.check_keywords(item, spider.name, item['pdf'])
+
+        # Remove the path from the value we are storing
+        full_item['pdf'] = os.path.basename(item['pdf'])
         full_item['hash'] = file_hash
         full_item['provider'] = spider.name
```

Comment:
again might worth exploring whether coupling the name with the provider can cause issues. for example for greater parallelisation it could be that we deployed two spiders for two different entrypoints in the same provider but with a different name. maybe add an issue if you agree

Reply:
still in the case of two spiders you might wanted to give them a different name or customise them differently for different entrypoints but we would not want the provider name to change

Reply:
Hum right, I see. Let's create an issue to think more about it 👍